lib/zendesk_api/middleware/response/parse_iso_dates.rb

```diff
@@ -19,7 +19,9 @@ def parse_dates!(value)
           case value
           when Hash then value.each { |key,element| value[key] = parse_dates!(element) }
           when Array then value.each_with_index { |element, index| value[index] = parse_dates!(element) }
-          when /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m then Time.parse(value)
+          when String
+            return Time.parse(value) if value.valid_encoding? && value =~ /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m
+            value
```

Comment:
The response itself, since I'm sure there's a bunch of bad data in the database. I couldn't find anything after a quick scan, and it looks like yajl doesn't actually care either when encoding. So maybe it is best that we just clean up all the strings in the client.